scrapers/StateOfTexas.mjs

```diff
@@ -5,28 +5,33 @@ export default ({ database, DataScraper }) => {
 	return DataScraper(database, 'StateOfTexas', async () => {
 		const data = [];
 		const baseUrl = 'https://api.tnris.org/api/v1';
-		const id = '2679b514-bb7b-409f-97f3-ee3879f34448';
-		const collections = `collections/${id}`;
-		const resources = `resources?collection_id=${id}`;
+		const collections = 'collections';
 		const collectionsRequest = await fetch(`${baseUrl}/${collections}`);
-		const collection = await collectionsRequest.json();
-		const resourcesRequest = await fetch(`${baseUrl}/${resources}`);
-		const resource = await resourcesRequest.json();
+		const collectionsData = await collectionsRequest.json();
+		const { results } = collectionsData;
 
-		const { publication_date, description, name } = collection;
-		const { results } = resource;
-		const created = moment(publication_date, 'YYYY-MM-DD').unix();
+		const promises = results.map(async collection => {
+			const { publication_date, description, name, id } = collection;
+			const resources = `resources?collection_id=${id}`;
+			const resourcesRequest = await fetch(`${baseUrl}/${resources}`);
+			const resourcesData = await resourcesRequest.json();
+			const { results: resourceResults } = resourcesData;
+			const created = moment(publication_date, 'YYYY-MM-DD').unix();
 
-		results.map(item => {
-			return data.push({
-				url: item.resource,
-				updated: 0,
-				created,
-				description,
-				name,
+			const collectionData = resourceResults.map(item => {
+				return {
+					url: item.resource,
+					updated: 0,
+					created,
+					description,
+					name,
+				};
 			});
+			data.push(...collectionData);
 		});
 
+		await Promise.all(promises);
```

Comment:
This shows great understanding of promises, however this isn't ideal. In this case we would be making 1000 requests all at once and we might hit nasty rate limits that break the scraper. I would prefer to keep this linear.

Reply:
I see. I assumed that making all the requests at once would mean better performance but didn't think of the API rate limits. Thank you!